### PR TITLE
cli: add `--ingress` flag to inject cmd

### PR DESCRIPTION
**Pull request by Pothulapati**
This PR adds a new inject flag called `--ingress` which when enabled
adds a new annotation i.e `linkerd.io/inject: ingress`.

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

**Review by adleong — APPROVED** [review]

**Review by kleimkuhler — APPROVED** [review]
Looks good; works well 👍

**Review by alpeb — COMMENTED** [review]
Not a big deal, and a bit of a TIOLI, but using this flag with `--manual` results in the annotation being set.
I mean, `linkerd inject --manual` doesn't add an inject annotation, but `linkerd inject --manual --ingress` does, which is a little inconsistent. And that will result in a little different code path in the webhook injector: it won't inject not because of the absence of the annotation but because of the presence of the proxy.

**Comment by adleong** [comment]
Really good catch, @alpeb!  I would expect `linkerd inject --manual --ingress` to set the environment variable in the proxy directly instead of setting the annotation.

**Comment by Pothulapati** [comment]
> I would expect linkerd inject --manual --ingress to set the environment variable in the proxy directly instead of setting the annotation.

I think the problem is that currently it does both with manual. which makes proxy-injector react to the event but skips injection by taking a different path i.e by recognising that there is already a proxy present.

This can be fixed by having an check during injection which removes this annotation if its a manual injection and the annotaiton value is set to `ingress`.

File: cli/cmd/inject.go [hunk 1]
```diff
@@ -188,6 +188,10 @@ func (rt resourceTransformerInject) transform(bytes []byte) ([]byte, []inject.Re
 	}
 
 	if len(rt.overrideAnnotations) > 0 {
+		// Do not set ingress annotation if its manual
+		if val, ok := rt.overrideAnnotations[k8s.ProxyInjectAnnotation]; ok && val == k8s.ProxyInjectIngress && rt.injectProxy {
+			delete(rt.overrideAnnotations, k8s.ProxyInjectAnnotation)
+		}
```
Comment (alpeb) [inline cli/cmd/inject.go:194]: You can move `delete(rt.overrideAnnotations, k8s.ProxyInjectAnnotation)` up under the existing `if injectProxy` block, and you don't need to check for the existence of the entry prior to deleting it :wink:

**Review by alpeb — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**